lib/dor/text_extraction/sqs_watcher.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Dor
+  module TextExtraction
+    # Watch an SQS queue for messages by long polling, allowing the consumer
+    # of this class to specify the queue name and the processing logic.
+    class SqsWatcher
+      attr_reader :logger, :queue_url, :role_arn, :visibility_timeout
+
+      def initialize(queue_url:, role_arn: nil, logger: Logger.new($stdout), visibility_timeout: nil)
+        @queue_url = queue_url
+        @role_arn = role_arn
+        @logger = logger
+        @visibility_timeout = visibility_timeout || default_visibility_timeout
+      end
+
+      def poll
+        logger.info("Starting indefinite long polling of #{queue_url}")
+        poller.poll(visibility_timeout:) do |sqs_msg|
+          yield sqs_msg
+        rescue StandardError => e
```

Comment:
It may be necessary to catch session expiration exceptions here and then run assume role to get an updated session token. Something to keep an eye on if things don't appear to get picked up.

Reply:
ah, yeah, hadn't read this far when i responded up there, but that's what i was thinking in part of my response to the token expiration concern you mentioned in `AwsProvider`.

Reply:
good news for complexity we don't have to maintain: i started the watcher daemon locally last night (22:50) before ending work for the day, queued and processed a speech to text job this afternoon (13:52) using my local speech-to-text docker container, and saw the daemon attempt to pick up the job 15 hours after it was originally started, with no auth related exception.  the attempt to process the done message by the poller failed, because my local common-accessioning isn't configured to talk to a live DSA or workflow-server-rails, but it was clear the message got picked up with no trouble.  so hopefully the ruby client handles token renewal for us?  bonus: i got to see the message get picked up again a couple minutes later, after the visibility timeout expired, since the exception handling prevented message deletion.